crates/primitives/src/stage/id.rs

```diff
@@ -0,0 +1,108 @@
+/// Stage IDs for all known stages.
+///
+/// For custom stages, use [`StageId::Other`]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[allow(missing_docs)]
```

Comment:
We should reuse this in reth stage etc, let's add strum impls so that clap can parse this from cli